botocore/signers.py

```diff
@@ -130,3 +134,275 @@ def get_auth(self, signing_name, region_name, signature_version=None):
             auth = cls(**kwargs)
             self._cache[key] = auth
             return auth
+
+    def generate_url(self, request_dict, expires_in=3600, region_name=None):
```

Comment:
What about generate_presigned_url, or something with presigned in the method name?  I think it makes it more clear exactly what kind of URL it's generating.